base/iterators.jl

```diff
@@ -1089,10 +1089,9 @@ end
 
 @inline peek(s::Stateful, sentinel=nothing) = s.nextvalstate !== nothing ? s.nextvalstate[1] : sentinel
 @inline iterate(s::Stateful, state=nothing) = s.nextvalstate === nothing ? nothing : (popfirst!(s), nothing)
-IteratorSize(::Type{Stateful{VS,T}} where VS) where {T} =
```

Comment:
I don't quite remember what the intent here was, but it may have been to filter out `HasSize`, since we don't pass that through (and can't really support).